File: src/test/java/no/nav/fo/veilarbregistrering/db/registrering/ManuellRegistreringRepositoryDbIntegrationTest.java
package no.nav.fo.veilarbregistrering.db.registrering;

import no.nav.fo.veilarbregistrering.db.MigrationUtils;
import no.nav.fo.veilarbregistrering.db.TransactionalTest;
import no.nav.fo.veilarbregistrering.registrering.BrukerRegistreringType;
import no.nav.fo.veilarbregistrering.registrering.manuell.ManuellRegistrering;
import no.nav.fo.veilarbregistrering.registrering.manuell.ManuellRegistreringRepository;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jdbc.core.JdbcTemplate;

import static org.junit.jupiter.api.Assertions.assertEquals;

@TransactionalTest
public class ManuellRegistreringRepositoryDbIntegrationTest {

    @Autowired
    private JdbcTemplate jdbcTemplate;

    @Autowired
    private ManuellRegistreringRepository manuellRegistreringRepository;

    @BeforeAll
    public void setup() {
        MigrationUtils.createTables(jdbcTemplate);
    }

    @Test
    public void hentManuellRegistrering(){

        String veilederIdent = "Z1234567";
        String veilederEnhetId = "1234";
        long registreringId = 1;
        BrukerRegistreringType registreringType = BrukerRegistreringType.ORDINAER;

        ManuellRegistrering manuellRegistrering = new ManuellRegistrering()
                .setRegistreringId(registreringId)
                .setBrukerRegistreringType(registreringType)
                .setVeilederIdent(veilederIdent)
                .setVeilederEnhetId(veilederEnhetId);

        long id = manuellRegistreringRepository.lagreManuellRegistrering(manuellRegistrering);

        manuellRegistrering.setId(id);

        ManuellRegistrering hentetRegistrering = manuellRegistreringRepository
                .hentManuellRegistrering(registreringId, registreringType);

        assertEquals(manuellRegistrering, hentetRegistrering);

    }

}

File: src/test/java/no/nav/fo/veilarbregistrering/db/registrering/ManuellRegistreringRepositoryDbIntegrationTest.kt
package no.nav.fo.veilarbregistrering.db.registrering

import no.nav.fo.veilarbregistrering.db.DatabaseConfig
import no.nav.fo.veilarbregistrering.db.RepositoryConfig
import no.nav.fo.veilarbregistrering.registrering.BrukerRegistreringType
import no.nav.fo.veilarbregistrering.registrering.manuell.ManuellRegistrering
import no.nav.fo.veilarbregistrering.registrering.manuell.ManuellRegistreringRepository
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
import org.springframework.boot.test.autoconfigure.jdbc.JdbcTest
import org.springframework.test.context.ContextConfiguration

@JdbcTest
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
@ContextConfiguration(classes = [RepositoryConfig::class, DatabaseConfig::class])
class ManuellRegistreringRepositoryDbIntegrationTest(

    @Autowired
    private val manuellRegistreringRepository: ManuellRegistreringRepository
) {

    @Test
    fun hentManuellRegistrering() {
        val veilederIdent = "Z1234567"
        val veilederEnhetId = "1234"
        val registreringId: Long = 1
        val registreringType = BrukerRegistreringType.ORDINAER
        val manuellRegistrering = ManuellRegistrering(
            registreringId,
            registreringType,
            veilederIdent,
            veilederEnhetId,
        )
        val id = manuellRegistreringRepository.lagreManuellRegistrering(manuellRegistrering)
        val manuellRegistrering2 = ManuellRegistrering(
            id,
            manuellRegistrering.registreringId,
            manuellRegistrering.brukerRegistreringType,
            manuellRegistrering.veilederIdent,
            manuellRegistrering.veilederEnhetId
        )
        val hentetRegistrering = manuellRegistreringRepository.hentManuellRegistrering(registreringId, registreringType)
        assertEquals(manuellRegistrering2, hentetRegistrering)
    }
}